x-pack/legacy/plugins/siem/public/components/truncatable_text/index.tsx

```diff
@@ -4,21 +4,24 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { EuiText } from '@elastic/eui';
 import styled from 'styled-components';
 
 /**
  * Applies CSS styling to enable text to be truncated with an ellipsis.
  * Example: "Don't leave me hanging..."
  *
- * Width is required, because CSS will not truncate the text unless a width is
- * specified.
+ * Note: Requires a parent container with a defined width or max-width.
  */
-export const TruncatableText = styled(EuiText)<{ width: string }>`
-  overflow: hidden;
-  text-overflow: ellipsis;
-  white-space: nowrap;
-  width: ${({ width }) => width};
-`;
 
+export const TruncatableText = styled.span`
+  &,
+  & * {
+    display: inline-block;
+    max-width: 100%;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    vertical-align: top;
```

Comment:
should this behavior be opt-in (so `TruncatableText` may be used in any context)?

Reply:
Is this in reference to the `vertical-align: top;`? If so, that's actually there to ensure the wrapping element maintains the same height as the container it is nested in. Without it, it assumes `baseline` as the default and adversely messes with the element height. It shouldn't prevent us from using anywhere.